rclcpp/include/rclcpp/executors/static_memory_executor.hpp

```diff
@@ -0,0 +1,133 @@
+// Copyright 2014 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCLCPP_RCLCPP_EXECUTORS_STATIC_MEMORY_EXECUTOR_HPP_
+#define RCLCPP_RCLCPP_EXECUTORS_STATIC_MEMORY_EXECUTOR_HPP_
+
+#include <cassert>
+#include <cstdlib>
+
+#include <rmw/rmw.h>
+#include <memory>
+#include <vector>
+
+#include <rclcpp/executor.hpp>
+#include <rclcpp/macros.hpp>
+#include <rclcpp/node.hpp>
+#include <rclcpp/utilities.hpp>
+#include <rclcpp/rate.hpp>
+
+#define MAX_SUBSCRIPTIONS 16
+#define MAX_SERVICES 8
+#define MAX_CLIENTS 16
+#define MAX_GUARD_CONDS 100
```

Comment:
At the very least we should make them overridable, maybe like this:

```c++
#ifndef MAX_SUBSCRIPTIONS
#define MAX_SUBSCRIPTIONS 16
#endif
```

The downside to the above is that when we build this into a library that might not be possible for the user to override at that point.

Also, since these are macros and therefore have no implicit name spacing, please prefix them with `RCLCPP_`.